Code/GraphMol/ChemReactions/ReactionPickler.cpp

```diff
@@ -146,15 +146,18 @@ void ReactionPickler::_pickle(const ChemicalReaction *rxn, std::ostream &ss,
   streamWrite(ss, BEGINREACTANTS);
   for (auto tmpl = rxn->beginReactantTemplates();
        tmpl != rxn->endReactantTemplates(); ++tmpl) {
-    MolPickler::pickleMol(tmpl->get(), ss);
+    auto props = rxn->df_needsInit ? PicklerOps::PropertyPickleOptions::NoProps :
```

Comment:
I'd suggest pulling the definition of `props` out of the for loop here so that you don't have to repeat yourself later on line 158. Unless you can think of a scenario where you would want to use different properties for the reactants and products.

Reply:
the are different, the products default to AtomProps  and the reactants default to no props.  After initialization we need to pickle private props as well.

This was really done in the hopes that I didn't have to change every regression test that makes a reaction pickle.  I also think that the act of pickling changes the molecule which is a bigger issue that I didn't want to deal with now.

Reply:
> they are different, the products default to AtomProps and the reactants default to no props. After initialization we need to pickle private props as well.

Ah right, missed that... sorry. This looks fine then